dot/network/block_announce.go

```diff
@@ -18,7 +18,7 @@ import (
 var errInvalidRole = errors.New("invalid role")
 var (
 	_ NotificationsMessage = &BlockAnnounceMessage{}
-	_ NotificationsMessage = &BlockAnnounceHandshake{}
+	_ Handshake            = (*BlockAnnounceHandshake)(nil)
```

Comment:
why the nil case?

Reply:
This is a check to make sure that BlockAnnounceHandshake implements Handshake interface. I think, Unlike the previous times, with nil, we end up not allocating any memory.